src/inpod.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright Istio Authors
```

Comment:
We have two data sources for "local workloads": WDS and the new UDS model.

Once we get the UID from the UDS, we start the proxy in the network namespace.
Now we can let the pod start (TBH not sure how we block it, but I think you said we do?).
Pod starts, sends an outbound request.
The proxy that is running is ONLY running for that workload, but we still (1) need to have the source workload from WDS and (2) try to guess certificates, etc based on source IP.

Inbound has similar logic.

This feels fishy in a few ways (but likely more I haven't thought of):
* If we are still doing IP based checks, we haven't really resolved the ip spoofing attacks, right?
* We block the pod running until the proxy starts, but it may not actually be ready to handle traffic since we haven't gotten the WDS response

I feel like I had more concerns but i was thinking about this late last night and didn't write it down...

WDYT? I get this PR is allow inpod and the old model, and the changes I am proposing are large and invasive. So I don't necessarily want to tackle them in this PR, if at all -- but would be good to understand the gaps now and the path to resolving them in the future?

Reply:
here are my thoughts:
https://github.com/yuval-k/ztunnel/tree/inpod-dest-id

Reply:
i planned to do this as a follow up; LMK what you think

Reply:
>We block the pod running until the proxy starts, but it may not actually be ready to handle traffic since we haven't gotten the WDS response

I think the answer for this concern is on_demand

Reply:
Note, that this also requires a change on the node agent side, as the uid from WDS is different from the uid in UDS - we may want to re-think this part